### PR TITLE
Added atomic dipole to test.py

### PR DESCRIPTION
**Pull request by hanao2**
Added atomic dipole to test.py

The current code did not cover this case.

### CODE REVIEW

File: deepmd/entrypoints/test.py [hunk 1]
```diff
@@ -933,20 +933,37 @@ def test_dipole(
...
+                (
+                    test_data["atomic_dipole"][:numb_test],
+                    dipole,
+                ),
+                axis=1,
+            )
+
+            np.savetxt(
+                detail_path.with_suffix(".out"),
+                pe,
```
Comment (njzjz) [inline deepmd/entrypoints/test.py:965]: `header` should be added here, like line 950.

File: deepmd/entrypoints/test.py [hunk 2]
```diff
@@ -933,20 +933,37 @@ def test_dipole(
...
+                header="data_x data_y data_z pred_x pred_y pred_z",
+            )
+        else:
+            detail_path = Path(detail_file)
+
+            pe = np.concatenate(
+                (
+                    test_data["atomic_dipole"][:numb_test],
+                    dipole,
```
Comment (njzjz) [inline deepmd/entrypoints/test.py:958]: `reshape` like line 941?

File: deepmd/entrypoints/test.py (hunk 1 above)
Comment (njzjz) [reply deepmd/entrypoints/test.py:965]: It seems that we do not need to write it twice. Just put these lines out of the `if...else`.

**Comment by hanao2** [comment]
Thanks for the comments. Today I went through the code and some of my output files; I noticed that the atomic dipole array (from 'atomic_dipole.npy') undergoes a reindexing when reading the test data. Therefore, for consistency and proper comparison (with the model prediction), we need to address this reindexing in the output 'detailed_file'. I will update the code shortly.

**Review by njzjz — APPROVED** [review]

**Pull request merged**